src/ui/components/App.tsx

```diff
@@ -34,27 +51,41 @@ const App: FunctionalComponent<Props> = ({ Monaco, esbuildWorker, astroWorker })
           filename = 'Component1';
         } else {
           const number = Number.parseInt(filename.replace(/^Component/, '')) + 1;
-          filename = `Component${number}`
+          filename = `Component${number}`;
         }
       }
     }
-    addTab(`---
+    addTab(
+      `---
 const name = "Component"
 ---
 
-<h1>Hello {name}</h1>`, 'astro', Uri.parse(`inmemory://model/src/components/${filename}.astro`))
-  }, [models])
+<h1>Hello {name}</h1>`,
+      'astro',
+      Uri.parse(`inmemory://model/src/components/${filename}.astro`)
+    );
+  }, [models]);
 
   const onSetTab = (model) => {
     console.log(model.uri.toString());
     setTab(model);
-  }
+  };
 
   return (
     <>
-      <Editor ref={editorRef} models={models} currentModel={currentModel} onAddTab={onAddTab} onSetTab={onSetTab} onRemoveTab={removeTab}  />
-      <Preview hasError={!!error} html={html} js={js} />
-      <StatusBar error={error} duration={duration}/>
+      <Menu currentTab={currentTab} setCurrentTab={setCurrentTab} />
+      <Editor
+        currentModel={currentModel}
+        currentTab={currentTab}
+        models={models}
+        onAddTab={onAddTab}
+        onRemoveTab={removeTab}
+        onSetTab={onSetTab}
+        ref={editorRef}
+      />
+      <Preview currentTab={currentTab} hasError={!!error} html={html} />
+      <JS currentTab={currentTab} hasError={!!error} code={js} />
+      <StatusBar error={error} duration={duration} />
```

Comment:
Had to rework the structure a bit. By adding a “Code” tab that lived alongside “Preview“ and ”JS”, it was easier to make everything a sibling at the top level.

I could have cheated by just aligning Code with the existing structure, but I felt that it was pretty early in the project to do something “cheating” like that. This method just uses pure CSS Grid and IMO is easier to reason about (the menu is in “Menu,” etc.).